src/js/components/invoice/InvoiceItemsModal.jsx

```diff
@@ -1,44 +1,323 @@
+        defaultMessage: 'Shipment Number',
+      },
+      budgetCode: {
+        type: LabelField,
+        label: 'react.invoice.budgetCode.label',
+        defaultMessage: 'Budget Code',
+      },
+      glCode: {
+        type: LabelField,
+        label: 'react.invoice.glCode.label',
+        defaultMessage: 'GL Code',
+      },
+      productCode: {
+        type: LabelField,
+        label: 'react.invoice.itemNumber.label',
+        defaultMessage: 'Item No',
+      },
+      description: {
+        type: LabelField,
+        label: 'react.invoice.description.label',
+        defaultMessage: 'Description',
+      },
+      quantity: {
+        type: LabelField,
+        label: 'react.invoice.quantity.label',
+        defaultMessage: 'Qty',
+      },
+      quantityToInvoice: {
+        type: TextField,
+        label: 'react.invoice.quantityToInvoice.label',
+        defaultMessage: 'Qty to Invoice',
+        fixedWidth: '140px',
+        attributes: {
+          type: 'number',
+        },
+        getDynamicAttr: ({
+          updateRow, values, rowIndex,
+        }) => ({
+          onChange: () => updateRow(values, rowIndex),
+        }),
+      },
+      uom: {
+        type: LabelField,
+        label: 'react.invoice.uom.label',
+        defaultMessage: 'UoM',
+      },
+      unitPrice: {
+        type: LabelField,
+        label: 'react.invoice.unitPrice.label',
+        defaultMessage: 'Unit Price',
+      },
+    },
+  },
+};
+
+function validate(values) {
+  const errors = {};
+  errors.invoiceItems = [];
+
+  _.forEach(values.invoiceItems, (item, key) => {
+    if (
+      item.checked &&
+      (
+        (_.toInteger(item.quantityToInvoice) > item.quantity) ||
+        _.toInteger(item.quantityToInvoice) < 0
+      )
+    ) {
+      errors.invoiceItems[key] = { quantityToInvoice: 'react.invoice.errors.quantityToInvoice.label' };
+    }
+  });
+
+  return errors;
+}
+
+const INITIAL_STATE = {
+  orderNumberOptions: [],
+  selectedOrderNumber: '',
+  selectedShipmentNumber: '',
+  selectedInvoiceItems: [],
+  formValues: { invoiceItems: [] },
+  sortOrder: 0,
+};
 
 /* eslint-disable */
 class InvoiceItemsModal extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = INITIAL_STATE;
+
+    this.onOpen = this.onOpen.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.selectRow = this.selectRow.bind(this);
+    this.updateRow = this.updateRow.bind(this);
   }
 
   onOpen() {
+    this.setState(INITIAL_STATE, () => {
+      this.fetchInvoiceItemCandidates();
+    });
   }
 
   onSave() {
+    this.props.showSpinner();
+    const { invoiceId } = this.props;
+    const { selectedInvoiceItems } = this.state;
+
+    const payload = {
+      invoiceItems: _.map(selectedInvoiceItems, (item, key) => ({
+        id: key,
+        quantityToInvoice: _.toInteger(item.quantityToInvoice),
+        sortOrder: _.toInteger(item.sortOrder),
```

Comment:
Is this sortOrder needed and saved anywhere? There is no such field in InvoiceItem.